src/scenes/gameuiscreen.ts

```diff
@@ -46,22 +50,37 @@ export class GameUiScreen extends Phaser.Scene {
         assert(this.drawnTilesContainer);
         this.drawnTilesContainer.removeAll(true);
         const myTiles: Nullable<Tile[]> = TileManager.getInHand();
-        assert(myTiles && this.uiBackground);
+        const session: Nullable<Session> = sessionManager.get();
+        const me: Nullable<Player> = PlayerManager.getMe();
+        const players: Nullable<Player[]> = PlayerManager.getAll();
+        assert(myTiles && this.uiBackground && session && players && me);
+        const playerCount: int = players.length;
 
         const nrTiles: int = myTiles.length;
         const tileSpacing: int =
             (this.uiBackground.width - nrTiles * 128) / (nrTiles + 1);
 
         for (let i: int = 0; i < nrTiles; i++) {
-            this.drawnTilesContainer.add(
-                this.add.image(
-                    this.uiBackground.getTopLeft().x +
-                        (tileSpacing + 128 / 2) +
-                        i * (128 + tileSpacing),
-                    ScreenHeight - 100,
-                    `tile${myTiles[i].type}`,
-                ),
+            const image: Phaser.GameObjects.Image = this.add.image(
+                this.uiBackground.getTopLeft().x +
+                    (tileSpacing + 128 / 2) +
+                    i * (128 + tileSpacing),
+                ScreenHeight - 100,
+                `tile${myTiles[i].type}`,
             );
+            if (session.turnIndex % playerCount === me.orderIndex) {
+                if (session.turnIndex === 0 && me.orderIndex === 0) {
+                    this.drawnTilesContainer.add(image);
+                } else {
+                    this.drawnTilesContainer.add(
+                        image.setAlpha(0.5).setTint(0xfefeee),
+                    );
+                }
+            } else {
+                this.drawnTilesContainer.add(
+                    image.setAlpha(0.4).setTint(0xfefeee),
+                );
+            }
         }
```

Comment:
Can we put this if statement in two function like 
private checkMyTurn(): boolean {
    return true if session.turnIndex % playerCount === me.orderIndex, else false
}

private checkStartTurn(){
    if (session.turnIndex === 0 && me.orderIndex === 0){ ... }
}

If a player wants to click on a tile or place a tile, first check if checkMyTurn() === true, only then allow action